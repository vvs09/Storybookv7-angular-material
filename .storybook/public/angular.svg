<svg width="188" height="200" viewBox="0 0 188 200" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M94 123.4H69.3L57.6 152.6H35.9H35.8L94 22.1V22.2V0L0.899994 33.2L15.1 156.3L94 200V123.4Z" fill="#808080"/>
<path d="M77 105.4H94V64.5L77 105.4Z" fill="#808080"/>
<g opacity="0.9">
<path d="M94 123.4H118.7L130.4 152.6H152.1H152.2L94 22.1V22.2V0L187.1 33.2L172.9 156.3L94 200V123.4Z" fill="#808080"/>
<path d="M111 105.4H94V64.5L111 105.4Z" fill="#808080"/>
</g>
</svg>
