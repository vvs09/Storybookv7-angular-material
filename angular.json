{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "newProjectRoot": "projects",
  "projects": {
    "storybook": {
      "projectType": "application",
      "schematics": {
        "@schematics/angular:component": {
          "style": "sass"
        }
      },
      "root": "",
      "sourceRoot": "src",
      "prefix": "app",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "outputPath": "dist/storybook",
            "index": "src/index.html",
            "main": "src/main.ts",
            "polyfills": [
              "zone.js"
            ],
            "tsConfig": "tsconfig.app.json",
            "inlineStyleLanguage": "sass",
            "assets": [
              "src/favicon.ico",
              "src/assets"
            ],
            "styles": [
              "@angular/material/prebuilt-themes/indigo-pink.css",
              "src/styles.sass"
            ],
            "scripts": []
          },
          "configurations": {
            "production": {
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "500kb",
                  "maximumError": "1mb"
                },
                {
                  "type": "anyComponentStyle",
                  "maximumWarning": "2kb",
                  "maximumError": "4kb"
                }
              ],
              "outputHashing": "all"
            },
            "development": {
              "buildOptimizer": false,
              "optimization": false,
              "vendorChunk": true,
              "extractLicenses": false,
              "sourceMap": true,
              "namedChunks": true
            }
          },
          "defaultConfiguration": "production"
        },
        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "configurations": {
            "production": {
              "buildTarget": "storybook:build:production"
            },
            "development": {
              "buildTarget": "storybook:build:development"
            }
          },
          "defaultConfiguration": "development"
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "buildTarget": "storybook:build"
          }
        },
        "test": {
          "builder": "@angular-devkit/build-angular:karma",
          "options": {
            "polyfills": [
              "zone.js",
              "zone.js/testing"
            ],
            "tsConfig": "tsconfig.spec.json",
            "inlineStyleLanguage": "sass",
            "assets": [
              "src/favicon.ico",
              "src/assets"
            ],
            "styles": [
              "@angular/material/prebuilt-themes/indigo-pink.css",
              "src/styles.sass"
            ],
            "scripts": []
          }
        },
        "storybook": {
          "builder": "@storybook/angular:start-storybook",
          "options": {
            "configDir": ".storybook",
            "browserTarget": "storybook:build",
            "compodoc": false,
            "port": 6006,
            "enableProdMode":true
          }
        },
        "build-storybook": {
          "builder": "@storybook/angular:build-storybook",
          "options": {
            "configDir": ".storybook",
            "browserTarget": "storybook:build",
            "compodoc": false,
            "outputDir": "storybook-static"
          }
        }
      }
    }
  }
}