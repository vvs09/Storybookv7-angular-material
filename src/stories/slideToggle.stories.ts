import type { Meta, StoryObj } from '@storybook/angular';
import { argsToTemplate, moduleMetadata } from '@storybook/angular';
import {
  MatSlideToggleModule,
  MatSlideToggle,
} from '@angular/material/slide-toggle';

const meta: Meta<MatSlideToggle> = {
  title: 'Components/Slide Toggle',
  component: MatSlideToggle,
  tags: ['autodocs'],
  argTypes: {
    color: {
      control: 'radio',
      options: ['accent', 'primary', 'warn'],
      description: 'Theme color palette for the component.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '-' },
        type: {
          summary: 'string',
        },
      },
    },
    ariaDescribedby: {
      control: 'text',
      description:
        'Used to set the aria-describedby attribute on the underlying input element.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '-' },
        type: {
          summary: 'string',
        },
      },
    },
    ariaLabel: {
      control: 'text',
      description:
        'Used to set the aria-label attribute on the underlying input element.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '-' },
        type: {
          summary: 'string',
        },
      },
    },
    ariaLabelledby: {
      control: 'text',
      description:
        'Used to set the aria-labelledby attribute on the underlying input element.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '-' },
        type: {
          summary: 'string',
        },
      },
    },
    checked: {
      control: 'boolean',
      description: 'Whether the slide-toggle element is checked or not.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '-' },
        type: {
          summary: 'boolean',
        },
      },
    },
    disableRipple: {
      control: 'boolean',
      description: 'Whether ripples are disabled.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '-' },
        type: {
          summary: 'boolean',
        },
      },
    },
    disabled: {
      control: 'boolean',
      description: 'Whether the component is disabled.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '-' },
        type: {
          summary: 'boolean',
        },
      },
    },
    id: {
      control: 'text',
      description:
        'A unique id for the slide-toggle input. If none is supplied, it will be auto-generated.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '-' },
        type: {
          summary: 'string',
        },
      },
    },
    labelPosition: {
      control: 'radio',
      options: ['after', 'before'],
      description:
        'Whether the label should appear after or before the slide-toggle. Defaults to after',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: 'after' },
        type: {
          summary: 'after | before',
        },
      },
    },
    name: {
      control: 'text',
      description:
        'Name value will be applied to the input element if present.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '-' },
        type: {
          summary: 'string',
        },
      },
    },
    required: {
      control: 'boolean',
      description: 'Whether the slide-toggle is required.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '-' },
        type: {
          summary: 'boolean',
        },
      },
    },
    buttonId: {
      control: 'text',
      description: 'Returns the unique id for the visual hidden button.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '-' },
        type: {
          summary: 'string',
        },
      },
    },
    defaults: {
      control: 'number',
      description: '',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '-' },
        type: {
          summary: 'MatSlideToggleDefaultOptions',
        },
      },
    },
    inputId: {
      control: 'text',
      description: 'Returns the unique id for the visual hidden input.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: '-' },
      },
    },
    toggleChange: {
      action: 'toggleChange',
      control: '-',
      description:
        'An event will be dispatched each time the slide-toggle input is toggled. This event is always emitted when the user toggles the slide toggle, but this does not mean the slide toggles value has changed.',
      table: {
        category: 'OUTPUTS',
        defaultValue: { summary: '-' },
        type: {
          summary: 'EventEmitter<MatSlideToggleChange>',
        },
      },
    },
    change: {
      action: 'change',
      control: '-',
      description:
        'An event will be dispatched each time the slide-toggle changes its value.',
      table: {
        category: 'OUTPUTS',
        defaultValue: { summary: '-' },
        type: {
          summary: 'EventEmitter<void>',
        },
      },
    },
  },
  decorators: [
    moduleMetadata({
      imports: [MatSlideToggleModule],
    }),
  ],
  render: (args: MatSlideToggle) => ({
    props: args,
    template: `<mat-slide-toggle ${argsToTemplate(
      args
    )}>Slide me!</mat-slide-toggle>
    `,
  }),
};

export default meta;
type Story = StoryObj<MatSlideToggle>;

export const Default: Story = {
  args: {
    ariaDescribedby: 'Silde Toggle',
    ariaLabel: 'Silde Toggle',
    ariaLabelledby: 'Silde Toggle',
    color: 'primary',
    checked: true,
    disableRipple: false,
    disabled: false,
    id: 'slide toggle storybook',
    labelPosition: 'after',
    name: 'Slide Toggle',
    required: false,
    buttonId: 'button id',
    // defaultColor: 'primary',
    inputId: 'input id',
  },
};

export const Warn: Story = {
  args: {
    ...Default.args,
    color: 'warn',
  },
};

export const Accent: Story = {
  args: {
    ...Default.args,
    color: 'accent',
  },
};

export const PositionBefore: Story = {
  args: {
    ...Default.args,
    labelPosition: 'before',
  },
};

export const Checked: Story = {
  args: {
    ...Default.args,
    checked: false,
  },
};

export const DisableRipple: Story = {
  args: {
    ...Default.args,
    disableRipple: true,
  },
};

export const Disable: Story = {
  args: {
    ...Default.args,
    disabled: true,
  },
};
