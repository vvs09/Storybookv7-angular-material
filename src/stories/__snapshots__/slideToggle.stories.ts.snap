// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Slide Toggle Accent smoke-test 1`] = `
<storybook-root ng-version="15.2.10">
  <mat-slide-toggle class="mat-mdc-slide-toggle mat-accent mat-mdc-slide-toggle-checked"
                    ng-reflect-color="accent"
                    ng-reflect-checked="true"
                    ng-reflect-disable-ripple="false"
                    ng-reflect-disabled="false"
                    ng-reflect-id="slide toggle storybook"
                    ng-reflect-label-position="after"
                    ng-reflect-name="Slide Toggle"
                    ng-reflect-required="false"
                    id="slide toggle storybook"
  >
    <div class="mdc-form-field">
      <button role="switch"
              type="button"
              class="mdc-switch mdc-switch--selected mdc-switch--checked"
              tabindex="0"
              id="slide toggle storybook-button"
              name="Slide Toggle"
              aria-labelledby="mat-mdc-slide-toggle-3-label"
              aria-checked="true"
      >
        <div class="mdc-switch__track">
        </div>
        <div class="mdc-switch__handle-track">
          <div class="mdc-switch__handle">
            <div class="mdc-switch__shadow">
              <div class="mdc-elevation-overlay">
              </div>
            </div>
            <div class="mdc-switch__ripple">
              <div mat-ripple
                   class="mat-ripple mat-mdc-slide-toggle-ripple mat-mdc-focus-indicator"
                   ng-reflect-trigger="[object HTMLButtonElement]"
                   ng-reflect-disabled="false"
                   ng-reflect-centered="true"
              >
              </div>
            </div>
            <div class="mdc-switch__icons">
              <svg viewbox="0 0 24 24"
                   class="mdc-switch__icon mdc-switch__icon--on"
              >
                <path d="M19.69,5.23L8.96,15.96l-4.23-4.23L2.96,13.5l6,6L21.46,7L19.69,5.23z">
                </path>
              </svg>
              <svg viewbox="0 0 24 24"
                   class="mdc-switch__icon mdc-switch__icon--off"
              >
                <path d="M20 13H4v-2h16v2z">
                </path>
              </svg>
            </div>
          </div>
        </div>
      </button>
      <label for="slide toggle storybook-button"
             id="mat-mdc-slide-toggle-3-label"
      >
        Slide me!
      </label>
    </div>
  </mat-slide-toggle>
</storybook-root>
`;

exports[`Components/Slide Toggle Checked smoke-test 1`] = `
<storybook-root ng-version="15.2.10">
  <mat-slide-toggle class="mat-mdc-slide-toggle mat-primary"
                    ng-reflect-color="primary"
                    ng-reflect-checked="false"
                    ng-reflect-disable-ripple="false"
                    ng-reflect-disabled="false"
                    ng-reflect-id="slide toggle storybook"
                    ng-reflect-label-position="after"
                    ng-reflect-name="Slide Toggle"
                    ng-reflect-required="false"
                    id="slide toggle storybook"
  >
    <div class="mdc-form-field">
      <button role="switch"
              type="button"
              class="mdc-switch mdc-switch--unselected"
              tabindex="0"
              id="slide toggle storybook-button"
              name="Slide Toggle"
              aria-labelledby="mat-mdc-slide-toggle-5-label"
              aria-checked="false"
      >
        <div class="mdc-switch__track">
        </div>
        <div class="mdc-switch__handle-track">
          <div class="mdc-switch__handle">
            <div class="mdc-switch__shadow">
              <div class="mdc-elevation-overlay">
              </div>
            </div>
            <div class="mdc-switch__ripple">
              <div mat-ripple
                   class="mat-ripple mat-mdc-slide-toggle-ripple mat-mdc-focus-indicator"
                   ng-reflect-trigger="[object HTMLButtonElement]"
                   ng-reflect-disabled="false"
                   ng-reflect-centered="true"
              >
              </div>
            </div>
            <div class="mdc-switch__icons">
              <svg viewbox="0 0 24 24"
                   class="mdc-switch__icon mdc-switch__icon--on"
              >
                <path d="M19.69,5.23L8.96,15.96l-4.23-4.23L2.96,13.5l6,6L21.46,7L19.69,5.23z">
                </path>
              </svg>
              <svg viewbox="0 0 24 24"
                   class="mdc-switch__icon mdc-switch__icon--off"
              >
                <path d="M20 13H4v-2h16v2z">
                </path>
              </svg>
            </div>
          </div>
        </div>
      </button>
      <label for="slide toggle storybook-button"
             id="mat-mdc-slide-toggle-5-label"
      >
        Slide me!
      </label>
    </div>
  </mat-slide-toggle>
</storybook-root>
`;

exports[`Components/Slide Toggle Default smoke-test 1`] = `
<storybook-root ng-version="15.2.10">
  <mat-slide-toggle class="mat-mdc-slide-toggle mat-primary mat-mdc-slide-toggle-checked"
                    ng-reflect-color="primary"
                    ng-reflect-checked="true"
                    ng-reflect-disable-ripple="false"
                    ng-reflect-disabled="false"
                    ng-reflect-id="slide toggle storybook"
                    ng-reflect-label-position="after"
                    ng-reflect-name="Slide Toggle"
                    ng-reflect-required="false"
                    id="slide toggle storybook"
  >
    <div class="mdc-form-field">
      <button role="switch"
              type="button"
              class="mdc-switch mdc-switch--selected mdc-switch--checked"
              tabindex="0"
              id="slide toggle storybook-button"
              name="Slide Toggle"
              aria-labelledby="mat-mdc-slide-toggle-1-label"
              aria-checked="true"
      >
        <div class="mdc-switch__track">
        </div>
        <div class="mdc-switch__handle-track">
          <div class="mdc-switch__handle">
            <div class="mdc-switch__shadow">
              <div class="mdc-elevation-overlay">
              </div>
            </div>
            <div class="mdc-switch__ripple">
              <div mat-ripple
                   class="mat-ripple mat-mdc-slide-toggle-ripple mat-mdc-focus-indicator"
                   ng-reflect-trigger="[object HTMLButtonElement]"
                   ng-reflect-disabled="false"
                   ng-reflect-centered="true"
              >
              </div>
            </div>
            <div class="mdc-switch__icons">
              <svg viewbox="0 0 24 24"
                   class="mdc-switch__icon mdc-switch__icon--on"
              >
                <path d="M19.69,5.23L8.96,15.96l-4.23-4.23L2.96,13.5l6,6L21.46,7L19.69,5.23z">
                </path>
              </svg>
              <svg viewbox="0 0 24 24"
                   class="mdc-switch__icon mdc-switch__icon--off"
              >
                <path d="M20 13H4v-2h16v2z">
                </path>
              </svg>
            </div>
          </div>
        </div>
      </button>
      <label for="slide toggle storybook-button"
             id="mat-mdc-slide-toggle-1-label"
      >
        Slide me!
      </label>
    </div>
  </mat-slide-toggle>
</storybook-root>
`;

exports[`Components/Slide Toggle Disable smoke-test 1`] = `
<storybook-root ng-version="15.2.10">
  <mat-slide-toggle class="mat-mdc-slide-toggle mat-primary mat-mdc-slide-toggle-checked"
                    ng-reflect-color="primary"
                    ng-reflect-checked="true"
                    ng-reflect-disable-ripple="false"
                    ng-reflect-disabled="true"
                    ng-reflect-id="slide toggle storybook"
                    ng-reflect-label-position="after"
                    ng-reflect-name="Slide Toggle"
                    ng-reflect-required="false"
                    id="slide toggle storybook"
  >
    <div class="mdc-form-field">
      <button role="switch"
              type="button"
              class="mdc-switch mdc-switch--selected mdc-switch--checked mdc-switch--disabled"
              tabindex="-1"
              disabled
              id="slide toggle storybook-button"
              name="Slide Toggle"
              aria-labelledby="mat-mdc-slide-toggle-7-label"
              aria-checked="true"
      >
        <div class="mdc-switch__track">
        </div>
        <div class="mdc-switch__handle-track">
          <div class="mdc-switch__handle">
            <div class="mdc-switch__shadow">
              <div class="mdc-elevation-overlay">
              </div>
            </div>
            <div class="mdc-switch__ripple">
              <div mat-ripple
                   class="mat-ripple mat-mdc-slide-toggle-ripple mat-mdc-focus-indicator"
                   ng-reflect-trigger="[object HTMLButtonElement]"
                   ng-reflect-disabled="true"
                   ng-reflect-centered="true"
              >
              </div>
            </div>
            <div class="mdc-switch__icons">
              <svg viewbox="0 0 24 24"
                   class="mdc-switch__icon mdc-switch__icon--on"
              >
                <path d="M19.69,5.23L8.96,15.96l-4.23-4.23L2.96,13.5l6,6L21.46,7L19.69,5.23z">
                </path>
              </svg>
              <svg viewbox="0 0 24 24"
                   class="mdc-switch__icon mdc-switch__icon--off"
              >
                <path d="M20 13H4v-2h16v2z">
                </path>
              </svg>
            </div>
          </div>
        </div>
      </button>
      <label for="slide toggle storybook-button"
             id="mat-mdc-slide-toggle-7-label"
      >
        Slide me!
      </label>
    </div>
  </mat-slide-toggle>
</storybook-root>
`;

exports[`Components/Slide Toggle DisableRipple smoke-test 1`] = `
<storybook-root ng-version="15.2.10">
  <mat-slide-toggle class="mat-mdc-slide-toggle mat-primary mat-mdc-slide-toggle-checked"
                    ng-reflect-color="primary"
                    ng-reflect-checked="true"
                    ng-reflect-disable-ripple="true"
                    ng-reflect-disabled="false"
                    ng-reflect-id="slide toggle storybook"
                    ng-reflect-label-position="after"
                    ng-reflect-name="Slide Toggle"
                    ng-reflect-required="false"
                    id="slide toggle storybook"
  >
    <div class="mdc-form-field">
      <button role="switch"
              type="button"
              class="mdc-switch mdc-switch--selected mdc-switch--checked"
              tabindex="0"
              id="slide toggle storybook-button"
              name="Slide Toggle"
              aria-labelledby="mat-mdc-slide-toggle-6-label"
              aria-checked="true"
      >
        <div class="mdc-switch__track">
        </div>
        <div class="mdc-switch__handle-track">
          <div class="mdc-switch__handle">
            <div class="mdc-switch__shadow">
              <div class="mdc-elevation-overlay">
              </div>
            </div>
            <div class="mdc-switch__ripple">
              <div mat-ripple
                   class="mat-ripple mat-mdc-slide-toggle-ripple mat-mdc-focus-indicator"
                   ng-reflect-trigger="[object HTMLButtonElement]"
                   ng-reflect-disabled="true"
                   ng-reflect-centered="true"
              >
              </div>
            </div>
            <div class="mdc-switch__icons">
              <svg viewbox="0 0 24 24"
                   class="mdc-switch__icon mdc-switch__icon--on"
              >
                <path d="M19.69,5.23L8.96,15.96l-4.23-4.23L2.96,13.5l6,6L21.46,7L19.69,5.23z">
                </path>
              </svg>
              <svg viewbox="0 0 24 24"
                   class="mdc-switch__icon mdc-switch__icon--off"
              >
                <path d="M20 13H4v-2h16v2z">
                </path>
              </svg>
            </div>
          </div>
        </div>
      </button>
      <label for="slide toggle storybook-button"
             id="mat-mdc-slide-toggle-6-label"
      >
        Slide me!
      </label>
    </div>
  </mat-slide-toggle>
</storybook-root>
`;

exports[`Components/Slide Toggle PositionBefore smoke-test 1`] = `
<storybook-root ng-version="15.2.10">
  <mat-slide-toggle class="mat-mdc-slide-toggle mat-primary mat-mdc-slide-toggle-checked"
                    ng-reflect-color="primary"
                    ng-reflect-checked="true"
                    ng-reflect-disable-ripple="false"
                    ng-reflect-disabled="false"
                    ng-reflect-id="slide toggle storybook"
                    ng-reflect-label-position="before"
                    ng-reflect-name="Slide Toggle"
                    ng-reflect-required="false"
                    id="slide toggle storybook"
  >
    <div class="mdc-form-field mdc-form-field--align-end">
      <button role="switch"
              type="button"
              class="mdc-switch mdc-switch--selected mdc-switch--checked"
              tabindex="0"
              id="slide toggle storybook-button"
              name="Slide Toggle"
              aria-labelledby="mat-mdc-slide-toggle-4-label"
              aria-checked="true"
      >
        <div class="mdc-switch__track">
        </div>
        <div class="mdc-switch__handle-track">
          <div class="mdc-switch__handle">
            <div class="mdc-switch__shadow">
              <div class="mdc-elevation-overlay">
              </div>
            </div>
            <div class="mdc-switch__ripple">
              <div mat-ripple
                   class="mat-ripple mat-mdc-slide-toggle-ripple mat-mdc-focus-indicator"
                   ng-reflect-trigger="[object HTMLButtonElement]"
                   ng-reflect-disabled="false"
                   ng-reflect-centered="true"
              >
              </div>
            </div>
            <div class="mdc-switch__icons">
              <svg viewbox="0 0 24 24"
                   class="mdc-switch__icon mdc-switch__icon--on"
              >
                <path d="M19.69,5.23L8.96,15.96l-4.23-4.23L2.96,13.5l6,6L21.46,7L19.69,5.23z">
                </path>
              </svg>
              <svg viewbox="0 0 24 24"
                   class="mdc-switch__icon mdc-switch__icon--off"
              >
                <path d="M20 13H4v-2h16v2z">
                </path>
              </svg>
            </div>
          </div>
        </div>
      </button>
      <label for="slide toggle storybook-button"
             id="mat-mdc-slide-toggle-4-label"
      >
        Slide me!
      </label>
    </div>
  </mat-slide-toggle>
</storybook-root>
`;

exports[`Components/Slide Toggle Warn smoke-test 1`] = `
<storybook-root ng-version="15.2.10">
  <mat-slide-toggle class="mat-mdc-slide-toggle mat-warn mat-mdc-slide-toggle-checked"
                    ng-reflect-color="warn"
                    ng-reflect-checked="true"
                    ng-reflect-disable-ripple="false"
                    ng-reflect-disabled="false"
                    ng-reflect-id="slide toggle storybook"
                    ng-reflect-label-position="after"
                    ng-reflect-name="Slide Toggle"
                    ng-reflect-required="false"
                    id="slide toggle storybook"
  >
    <div class="mdc-form-field">
      <button role="switch"
              type="button"
              class="mdc-switch mdc-switch--selected mdc-switch--checked"
              tabindex="0"
              id="slide toggle storybook-button"
              name="Slide Toggle"
              aria-labelledby="mat-mdc-slide-toggle-2-label"
              aria-checked="true"
      >
        <div class="mdc-switch__track">
        </div>
        <div class="mdc-switch__handle-track">
          <div class="mdc-switch__handle">
            <div class="mdc-switch__shadow">
              <div class="mdc-elevation-overlay">
              </div>
            </div>
            <div class="mdc-switch__ripple">
              <div mat-ripple
                   class="mat-ripple mat-mdc-slide-toggle-ripple mat-mdc-focus-indicator"
                   ng-reflect-trigger="[object HTMLButtonElement]"
                   ng-reflect-disabled="false"
                   ng-reflect-centered="true"
              >
              </div>
            </div>
            <div class="mdc-switch__icons">
              <svg viewbox="0 0 24 24"
                   class="mdc-switch__icon mdc-switch__icon--on"
              >
                <path d="M19.69,5.23L8.96,15.96l-4.23-4.23L2.96,13.5l6,6L21.46,7L19.69,5.23z">
                </path>
              </svg>
              <svg viewbox="0 0 24 24"
                   class="mdc-switch__icon mdc-switch__icon--off"
              >
                <path d="M20 13H4v-2h16v2z">
                </path>
              </svg>
            </div>
          </div>
        </div>
      </button>
      <label for="slide toggle storybook-button"
             id="mat-mdc-slide-toggle-2-label"
      >
        Slide me!
      </label>
    </div>
  </mat-slide-toggle>
</storybook-root>
`;
